### PR TITLE
Force re-execution of Partytown's head snippet on view transitions

### PR DESCRIPTION
**Pull request by martrapp**
## Changes

Just before the view transition swap(), remove the Partytown `<head>` snippet.
This forces a re-execution on the target page.

## Testing

Manually tested

## Docs
n.a. / bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
Looks good!

**Review by matthewp — APPROVED** [review]

**Comment by matthewp** [comment]
Does this fix an open issue?

File: packages/integrations/partytown/src/index.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export default function createPlugin(options?: PartytownOptions): AstroIntegrati
 	let partytownSnippetHtml: string;
 	const partytownEntrypoint = resolve('@builder.io/partytown/package.json');
 	const partytownLibDirectory = path.resolve(partytownEntrypoint, '../lib');
+	const SELF_DESTRUCT_ON_VIEW_TRANSITION = `;((d,s)=>(s=d.currentScript,d.addEventListener('astro:before-swap',()=>s?.remove(),{once:true})))(document);`
```
Comment (lilnasy) [inline packages/integrations/partytown/src/index.ts:24]: The logic seems correct. Is this an IIFE to save on code size?

**Comment by martrapp** [comment]
> Does this fix an open issue?

@OliverSpeir had an inspiring (pun intended) example which got stuck in view transitions or didn't work at all once view transitions were enabled. This, and the discussions with him, helped a lot as minimal reproduction, but to my knowledge, we currently do not have an official issue.

File: packages/integrations/partytown/src/index.ts (hunk 1 above)
Comment (martrapp) [reply packages/integrations/partytown/src/index.ts:24]: Yes, I started with Typescript, chrunched it a bit and then had fun with it. Mainly it fits the style of the original Partytown snippet.

Comment (martrapp) [reply packages/integrations/partytown/src/index.ts]: Also I needed a var to save currentScript for later. Always better than storing in window.

Comment (lilnasy) [reply packages/integrations/partytown/src/index.ts]: It's pretty neat!

**Review by lilnasy — APPROVED** [review]
Bit of a hack but that's okay because that's what partytown itself is.
Looks great to me!

**Comment by martrapp** [comment]
> Bit of a hack but that's okay because that's what partytown itself is.

I have not yet found another part for Code Golf in the codebase. 
In the end, we all prefer to have maintainable code ;-)

**Pull request merged**